src/backend/distributed/executor/executor.c

```diff
@@ -1821,6 +1830,15 @@ ReceiveResults(WorkerSession *session, bool storeRows)
 			scanState->customScanState.ss.ps.ps_ResultTupleSlot->tts_tupleDescriptor;
 		attributeInputMetadata = TupleDescGetAttInMetadata(tupleDescriptor);
 		expectedColumnCount = tupleDescriptor->natts;
+	
+		if (scanState->tuplestorestate == NULL)
+		{
+			bool randomAccess = true;
+			bool interTransactions = false;
+			scanState->tuplestorestate =
+				tuplestore_begin_heap(randomAccess, interTransactions, work_mem);
```

Comment:
I've used the same argument values as `StoreQueryResult()` in `multi_router_executor.c`.

Reply:
I've been wondering, is there a way to remove `ScanState` from the executor and instead keep a tuple store and descriptor in the `DistributedExecution`?

It would be useful to have a function that can just write to a tuple store supplied by the caller instead of having to supply a whole `ScanState`. That way, it becomes easier to call `ExecuteTaskListExtended` from other contexts (e.g. a UDF).

Reply:
> I've been wondering, is there a way to remove `ScanState` from the executor and instead keep a tuple store and descriptor in the `DistributedExecution`?
> 
> It would be useful to have a function that can just write to a tuple store supplied by the caller instead of having to supply a whole `ScanState`. That way, it becomes easier to call `ExecuteTaskListExtended` from other contexts (e.g. a UDF).

I think that makes sense. I made the changes.